So 91

hashemJaber opened the pull request:
SO-91
- Added Tasks component 
- Added Tasks edit component
- Added client apis to select, add, patch tasks
- Added Tasks CRU components
- Added Task members CRUD operations

jvniorrr committed:
SO-94

- Specific projects page template
- Member invites setup
- Member card component
- Project Attachment placeholder setup, needs connection to database and storage of images with URLS now

hashemJaber committed:
SO-91
-Added Tasks component
- Added Tasks edit component
-Added client apis to select, add, patch tasks
- Added Tasks CRU components

hashemJaber committed:
SO-91 - Added Tasks component
- Added Tasks edit component
- Added client apis to select, add, patch tasks
- Added Tasks CRU components
- Added Task members CRUD operations

hashemJaber committed:
SO-91 Added Tasks component
Added Tasks edit component
Added client apis to select, add, patch tasks and task_members
Added Tasks CRUD components for tasks
Added Task members CRUD operations for task to members relationships
Added Validation between user and admin and setted up security policies on SupaBase

jvniorrr commented on src/components/projects/ProjectTasks/AddNewTask/AddNewTask.tsx:
If file no longer used lets remove this one or the other newTask.tsx

jvniorrr commented on src/components/projects/ProjectTasks/ContextProvider/index.tsx:
```diff
@@ -0,0 +1,109 @@
+	};
+
+	const updateTask = async (task: ITasks) => {
+		console.log('updating task:', task);
```
remove line

jvniorrr commented on src/components/projects/ProjectTasks/ContextProvider/index.tsx:
```diff
@@ -0,0 +1,109 @@
+			console.error('Error updating task:', error);
+		}
+
+		console.log('data:', data);
```
remove console.log statements

jvniorrr commented on src/components/projects/ProjectTasks/NewTask/index.tsx:
```diff
@@ -0,0 +1,359 @@
+
+	const [taskMembers, setTaskMembers]: any = useState([]);
+
+	const [members, setMembers] = useState([
```
Remove this if not using the new portion here the dropdown selection of members.

hashemJaber replied:
so this section is not used for drop down selection of members but it is used in other areas of the code

jvniorrr commented on src/components/projects/ProjectTasks/TaskCard/index.tsx:
```diff
@@ -0,0 +1,147 @@
+	const [members, setMembers] = useState<string[]>();
+
+	useEffect(() => {
+		const fetchMembers = async () => {
```
For now leave but lets leave comment in Jira informing of this as this will cause ton of API calls as well.

hashemJaber replied:
roger that

jvniorrr commented on src/lib/actions/client.ts:
```diff
@@ -133,3 +133,279 @@ export async function removeProjectMember(proj_id: string, user_id: string) {
 	return true;
 }
+
+export async function updateTask(task: ITasks) {
```
remove function before merging to. main. thoi update task accessing from `tasktmp` table.

jvniorrr commented on src/lib/actions/client.ts:
```diff
@@ -133,3 +133,279 @@ export async function removeProjectMember(proj_id: string, user_id: string) {
+		.eq('task_id', task.id);
+
+	if (error) {
+		alert(JSON.stringify(error));
```
do not push code with alerts preferably.

jvniorrr replied:
change to console.error(...)

jvniorrr commented on src/components/projects/ProjectTasks/client.tsx:
```diff
@@ -0,0 +1,542 @@
+					role={role}
+					setRefresh={() => {}}
+					dismiss={() => {}}
+					org_id={'d'}
```
org_id={''} ??? unsure there supposed to be different value here?

jvniorrr reviewed (changes requested):
Some changes to be made :). Nice work.

hashemJaber committed:
SO-91
- Added Tasks component
- Added Tasks edit component
- Added client apis to select, add, patch tasks
- Added Tasks CRU components
- Added Task members CRUD operations
